db/hints/resource_manager.hh

```diff
@@ -119,10 +120,45 @@ class resource_manager {
     const size_t _min_send_hint_budget;
     seastar::named_semaphore _send_limiter;
 
+    seastar::named_semaphore _operation_lock;
     space_watchdog::shard_managers_set _shard_managers;
     space_watchdog::per_device_limits_map _per_device_limits_map;
     space_watchdog _space_watchdog;
 
+    shared_ptr<service::storage_proxy> _proxy_ptr;
+    shared_ptr<gms::gossiper> _gossiper_ptr;
+    shared_ptr<service::storage_service> _ss_ptr;
+
+    enum class state {
+        running,
+        replay_allowed,
+    };
+    using state_set = enum_set<super_enum<state,
+        state::running,
+        state::replay_allowed>>;
+
```

Comment:
Shouldn't stop() be adjusted to consider the state?

Reply:
I'm not sure what you mean. `resource_manager::stop()` calls `unset_running()`, so the running bit is cleared when we stop.